File: 0560-subarray-sum-equals-k/0560-subarray-sum-equals-k.java
class Solution {
    public int subarraySum(int[] nums, int k) {
        HashMap<Integer, Integer> map = new HashMap<>();
        map.put(0, 1);

        int ans = 0;

        int prefixSum = 0;
        for(int num: nums){
            prefixSum += num;
            if(map.containsKey(prefixSum - k)){
                ans += map.get(prefixSum - k);
            }
            map.put(prefixSum, map.getOrDefault(prefixSum, 0) + 1);
        }

        return ans;
    }
}

File: 0560-subarray-sum-equals-k/0560-subarray-sum-equals-k.py
class Solution:
    def subarraySum(self, nums: List[int], k: int) -> int:
        prefixSum = 0
        prefixSumMap = {0: 1}
        result = 0

        for num in nums:
            prefixSum += num
            if prefixSumMap.get(prefixSum - k) != None:
                result += prefixSumMap.get(prefixSum - k)
            
            prefixSumMap[prefixSum] = prefixSumMap.get(prefixSum, 0) + 1
        
        return result

        